### PR TITLE
Fix toggling of component variation details that lack names

### PR DESCRIPTION
**Pull request by contolini**
"Show details" buttons don't work if their corresponding variation lacks a name. This PR captures a `variation_slug` variable that is created by concatenating all the variation fields (instead of just the variation name) to ensure a slug is generated.

See GHE/el-camino/issues/237#issuecomment-252410

## Additions

- `variation_slug` variable that consists of the first 40 characters of every variation field concatenated.
- Browser test that validates the show/hide details button on every single component page.

## Removals

- Removed all arrow functions from our browser tests per [Mocha recommendation](https://mochajs.org/#arrow-functions).

## Changes

- Some variation names and descriptions were still rendering empty divs so I further improved our verification of their existence by stripping whitespace with the `strip` filter.

## Testing

1. Clicking "show details" under the "Print headings" section should properly show the details on the [PR preview website](https://deploy-preview-507--cfpb-design-system.netlify.app/design-system/components/headings#print-headings-1) while on [production](https://cfpb.github.io/design-system/components/headings#print-headings-1) it does not.
1. Cloud browser tests should pass below.
1. `yarn test:browser` should pass locally.

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: docs/_includes/variation-content.html [hunk 1]
```diff
@@ -40,62 +40,78 @@ <h2 class="o-variation_group_title">{{ variation_group.variation_group_name }}</
...
+                <h3>{{ variation_name }}</h3>
             </div>
-            {% endif %}
+            {% endunless %}
 
-            {% if variation.variation_description %}
+            {% unless variation_description == '' %}
             <div class="m-variation_description">
-                {{ variation.variation_description | markdownify }}
+                {{ variation_description | markdownify }}
```
Comment (anselmbradford) [inline docs/_includes/variation-content.html:55]: Why not apply all the filters when the variable is created on L:44? Will it ever be used without markdownify?

File: test/browser/docs/netlify-cms.js [hunk 2]
```diff
@@ -69,7 +69,7 @@ describe( 'Netlify CMS', () => {
       expect( previewPane ).toHaveTextContaining( 'component pages are the best' );
     } );
 
-    it( 'should support switching between the various "show details" tabs', () => {
+    it( 'should support switching between the various "show details" tabs', function() {
```
Comment (anselmbradford) [inline test/browser/docs/netlify-cms.js:72]: Why is it necessary to strip out the arrow functions? Was there a problem with scope?

Comment (anselmbradford) [reply test/browser/docs/netlify-cms.js:72]: Nevermind, I see the mocha note.

File: docs/_includes/variation-content.html (hunk 1 above)
Comment (contolini) [reply docs/_includes/variation-content.html:55]: Good call. 👍

**Pull request merged**